site_scons/env_modules.py

```diff
@@ -87,6 +87,17 @@ def _mpi_module(self, mpi):
         load = []
         unload = []
 
+        if mpi in self._mpi_map:
+            for key, value in self._mpi_map.items():
+                if key == mpi:
+                    load = value
+                else:
+                    unload += value
+        else:
+            # Support forcing specific mpi modules
+            load = list(mpi)
+            unload = [item for value in self._mpi_map.values() for item in value if item != mpi]
+
         for key, value in self._mpi_map.items():
```

Comment:
@dinghwah I am looking to use this code in one of my PR and I think that lines 101:105 need to be removed.